src/System.Collections/src/System/Collections/Generic/SortedSet.cs

```diff
@@ -1132,13 +1132,14 @@ internal virtual void IntersectWithEnumerable(IEnumerable<T> other)
                 if (Contains(item))
                 {
                     toSave.Add(item);
+                    DoRemove(item);
                 }
             }
 
-            if (toSave.Count < Count)
+            Clear();
+            foreach (T item in toSave)
             {
-                Clear();
-                AddAllElements(toSave);
+                AddIfNotPresent(item);
```

Comment:
I only looked at https://github.com/dotnet/corefx/blob/master/src/System.Collections/tests/Generic/SortedSet/SortedSet.Generic.Tests.cs#L14 and I could not find single test for `UnionWith` method to check what arguments combination is tested. Perhaps I looked at the wrong file